src/cli/mod.rs

```diff
@@ -169,22 +169,107 @@ pub enum Command {
+        self.inner.locked
+    }
+}
+
+// Automatic validation when converting from raw args
+impl TryFrom<LockFileUsageArgsRaw> for LockFileUsageArgs {
+    type Error = miette::Error;
+
+    fn try_from(raw: LockFileUsageArgsRaw) -> miette::Result<Self> {
+        if raw.frozen && raw.locked {
+            miette::bail!("the argument '--locked' cannot be used with '--frozen'");
+        }
+        Ok(LockFileUsageArgs { inner: raw })
+    }
+}
+
+// For clap flattening - this provides automatic validation
+impl clap::FromArgMatches for LockFileUsageArgs {
+    fn from_arg_matches(matches: &clap::ArgMatches) -> Result<Self, clap::Error> {
+        let raw = LockFileUsageArgsRaw::from_arg_matches(matches)?;
+        raw.try_into().map_err(|e: miette::Error| {
+            clap::Error::raw(clap::error::ErrorKind::ArgumentConflict, e.to_string())
+        })
+    }
+
+    fn update_from_arg_matches(&mut self, matches: &clap::ArgMatches) -> Result<(), clap::Error> {
+        *self = Self::from_arg_matches(matches)?;
+        Ok(())
+    }
+}
+
+impl clap::Args for LockFileUsageArgs {
+    fn augment_args(cmd: clap::Command) -> clap::Command {
+        LockFileUsageArgsRaw::augment_args(cmd)
+    }
+
+    fn augment_args_for_update(cmd: clap::Command) -> clap::Command {
+        LockFileUsageArgsRaw::augment_args_for_update(cmd)
+    }
+}
+
+impl From<LockFileUsageArgs> for crate::environment::LockFileUsage {
+    fn from(value: LockFileUsageArgs) -> Self {
+        if value.frozen() {
+            Self::Frozen
+        } else if value.locked() {
+            Self::Locked
+        } else {
+            Self::Update
+        }
+    }
+}
+
+/// Configuration for lock file usage, used by LockFileUpdateConfig
+#[derive(Parser, Debug, Default, Clone)]
 pub struct LockFileUsageConfig {
     /// Install the environment as defined in the lockfile, doesn't update
     /// lockfile if it isn't up-to-date with the manifest file.
-    #[clap(long, conflicts_with = "locked", env = "PIXI_FROZEN", help_heading = consts::CLAP_UPDATE_OPTIONS)]
+    #[clap(long, env = "PIXI_FROZEN", help_heading = consts::CLAP_UPDATE_OPTIONS)]
     pub frozen: bool,
     /// Check if lockfile is up-to-date before installing the environment,
     /// aborts when lockfile isn't up-to-date with the manifest file.
-    #[clap(long, conflicts_with = "frozen", env = "PIXI_LOCKED", help_heading = consts::CLAP_UPDATE_OPTIONS)]
+    #[clap(long, env = "PIXI_LOCKED", help_heading = consts::CLAP_UPDATE_OPTIONS)]
     pub locked: bool,
 }
 
 impl From<LockFileUsageConfig> for crate::environment::LockFileUsage {
     fn from(value: LockFileUsageConfig) -> Self {
+        // Validate that both frozen and locked are not true simultaneously
+        if value.frozen && value.locked {
+            // Since this is a From implementation, we can't return an error
+            // This should be caught earlier in argument parsing
+            panic!("Both --frozen and --locked cannot be used together");
```

Comment:
This should return a Result so the message can be handled properly. Now this throws a panic.:
```
➜ pixi install --frozen --locked

thread 'main2' panicked at src/cli/mod.rs:270:13:
Both --frozen and --locked cannot be used together
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

thread 'main' panicked at src/main.rs:49:10:
Tokio executor failed, was there a panic?: Any { .. }
```